### PR TITLE
Query visualization cleanup

### PR DESCRIPTION
**Pull request by kdoh**
While working on #2931 I found it incredibly hard to parse the query visualization component which hadn't been given a good look in a while. This cleans the component up a bit and brings it up to date with some of our general conventions  in anticipation of #2931 and some other features like next questions that will be coming down the pipe soon.

### CODE REVIEW

**Comment by kdoh** [comment]
@tlrobinson I think I'm pretty close to done with this pending your thoughts and some prop type additions so if you have a second to look at it that'd be great.

File: frontend/src/metabase/query_builder/QueryVisualization.jsx [hunk 1]
```diff
@@ -206,168 +205,87 @@ export default class QueryVisualization extends Component {
...
                 <div className={visualizationClasses}>
                     {viz}
                 </div>
             </div>
         );
     }
 }
+
+
+const VisualizationResult = ({card, isObjectDetail, lastRunDatasetQuery, result, ...rest}) => {
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/QueryVisualization.jsx]: Any reason these aren't in a separate file like the others?

File: frontend/src/metabase/query_builder/VisualizationError.jsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      <span className="QueryError-adminEmail">
+          <a className="no-decoration" href={`mailto:${adminEmail}`}>
+              {adminEmail}
+          </a>
+      </span>
+  )
+}
+
+class VisualizationError extends Component {
+  constructor (props) {
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/VisualizationError.jsx]: You can leave out the constructor if all it does it call `super()`

File: frontend/src/metabase/query_builder/VisualizationError.jsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      super(props)
+  }
+
+  showDetailError() {
+      if (this._detailErrorLink && this._detailErrorBody ) {
+          findDOMNode(this._detailErrorLink).style.display = "none";
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/VisualizationError.jsx]: This is very un-React-like but I know it's not new code so it's fine if you want to leave it.

File: frontend/src/metabase/query_builder/QueryVisualization.jsx (hunk 1 above)
Comment (kdoh) [inline frontend/src/metabase/query_builder/QueryVisualization.jsx]: Mostly just for the convenience of working on it while refactoring. Planning on moving them out into their own files.

File: frontend/src/metabase/query_builder/VisualizationError.jsx (hunk 2 above)
Comment (kdoh) [inline frontend/src/metabase/query_builder/VisualizationError.jsx]: Hah yeah, force of habit :P

File: frontend/src/metabase/query_builder/VisualizationError.jsx (hunk 3 above)
Comment (kdoh) [inline frontend/src/metabase/query_builder/VisualizationError.jsx]: Yeah its pretty gross.  I'll add a ``// TODO: warcrime `` note so we can take a look at changing it or see if its trivial to change.

**Comment by kdoh** [comment]
@tlrobinson Made the changes you suggested.

**Comment by tlrobinson** [comment]
LGTM.

You mentioned wanting to fix up the prop types. It looks like there's a component or two missing, if you care.

Also, I've been trying to add `/* eslint "react/prop-types": "warn" */` to the top of files when I'm cleaning up prop types to get ESLint to validate they're all there. Eventually we should just enable that globally.
